### PR TITLE
[City gates] Better handling of the stat xml and inserts missing streets

### PR DESCRIPTION
**Pull request by NicEastvillage**
- Now doesn't require all edges to have a street already
- Inserts missing streets and keeps streets already in the stat-file
- Doesn't crash if the `streets` element is not present

### CODE REVIEW

File: perlin.py [hunk 1]
```diff
@@ -77,16 +76,27 @@ def apply_network_noise(net: sumolib.net.Net, xml: ElementTree, scale: float, oc
...
+    streets = xml.find("streets")
+    if streets is None:
+        streets = ET.SubElement(xml.getroot(), "streets")
+
+    # Some edges might already have a street, so we want to ignore those
+    known_streets = {street.attrib["edge"]: street for street in streets.findall("street")}
+
+    for edge in net.getEdges():
+        eid = edge.getID()
+        if eid not in known_streets:
```
Comment (falkecarlsen) [inline perlin.py:88]: So if an edge is not associated with a known street, we won't apply noisemap population and industry to it? 

I believe we should apply the noisemap to each street-element.

**Review by falkecarlsen — CHANGES_REQUESTED** [review]

File: perlin.py (hunk 1 above)
Comment (NicEastvillage) [reply perlin.py:88]: But then we override anything that the user have put in the stats file manually.

Comment (NicEastvillage) [reply perlin.py:88]: What we could do is adding an option for it. To override or not to override. But let's add that later.

Comment (falkecarlsen) [reply perlin.py:88]: Opened #10 as a response to this.

**Pull request merged**